test/directives/align_array_type.f90

```diff
@@ -0,0 +1,157 @@
+! RUN: %flang -O0 -S -emit-llvm %s -o - | FileCheck %s
+
+! CHECK: %struct[[BLOCK1:\.BSS[0-9]+]] = type <{ [356 x i8] }>
+! CHECK: %struct[[BLOCK2:\.BSS[0-9]+]] = type <{ [612 x i8] }>
+! CHECK: %struct[[BLOCK3:\.BSS[0-9]+]] = type <{ [1124 x i8] }>
+! CHECK: %struct[[BLOCK4:\.BSS[0-9]+]] = type <{ [2148 x i8] }>
+! CHECK: %struct[[BLOCK5:_module_align_[0-9]+_]] = type <{ [228 x i8] }>
+! CHECK: @[[BLOCK1]] = internal global %struct[[BLOCK1]] zeroinitializer, align 256
+! CHECK: @[[BLOCK2]] = internal global %struct[[BLOCK2]] zeroinitializer, align 512
+! CHECK: @[[BLOCK3]] = internal global %struct[[BLOCK3]] zeroinitializer, align 1024
+! CHECK: @[[BLOCK4]] = internal global %struct[[BLOCK4]] zeroinitializer, align 2048
+! CHECK: @[[BLOCK5]] = common global %struct[[BLOCK5]] zeroinitializer, align 128
+
+module module_align
+implicit none
+
+    !dir$ align 0x80
```

Comment:
I'm finding this use of hex slightly awkward. The Fortran world is populated by mathematicians and physicists, they don't usually reason in hex numbers. Apart from `!pgi$` pragmas which control the xflags directly, our test cases don't seem to use it like that. Can decimals be used here (as well or instead)?

Reply:
Hi @pawosm-arm, 

> I'm finding this use of hex slightly awkward. The Fortran world is populated by mathematicians and physicists, they don't usually reason in hex numbers. Apart from `!pgi$` pragmas which control the xflags directly, our test cases don't seem to use it like that. Can decimals be used here (as well or instead)?

Thanks for your explanation.

This align pragma supports both hexadecimal and decimal formats. In the patch I use gtok() and `itok` to get alignment value, and gtok() can parse both hexadecimal and decimal formats.

I will refactor the test cases into decimal format as you and @kiranchandramohan  suggested.